subprojects/emcee/worker/src/test/kotlin/com/avito/emcee/worker/rest/HttpServerTest.kt

```diff
@@ -6,47 +6,73 @@ import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import java.net.ServerSocket
+import java.util.concurrent.Executors
 
 class HttpServerTest {
 
-    private val server: HttpServer = HttpServer.Builder()
-        .addHandler(ProcessingBucketsRequestHandler())
-        .debug(false)
-        .build()
-    private val client: SimpleClient = SimpleClient()
-    private val defaultPort = 8080
+    private var port: Int? = null
+    private val client = SimpleClient()
+    private val server = createServer()
 
     @BeforeEach
     fun before() {
-        server.start(defaultPort)
-        client.start("127.0.0.1", defaultPort)
+        port = findAvailablePort()
+        Executors.newSingleThreadExecutor().submit { server.start(port!!) }
```

Comment:
Correct me, but it looks like `HttpServer.start` should block the main thread and it's only bothers in tests.
That's my workaround